app/Http/Controllers/Api/V1/UserController.php

```diff
@@ -512,6 +528,11 @@ public function update(Request $request, int $id): JsonResponse
             $input = $request->all();
 
             $user = User::where('id', $id)->first();
+
+            if (!Gate::allows('update', $user)) {
```

Comment:
Just a note, I'd put this above the db call, waste of cpu if there is no allowance there.

Reply:
in this example, the gate needs to know the userId as well as the user group - so cant avoid a DB call to get this information